src/sql/query.rs

```diff
@@ -1621,6 +1639,15 @@ fn plan_function<'a>(
             bail!("aggregate functions are not allowed in {}", ecx.name);
         }
     } else {
+        if sql_func.over.is_some() {
+            bail!("OVER specified but {}() is not a window function", ident);
+        }
+        if sql_func.filter.is_some() {
+            bail!(
+                "FILTER specified but {}() is not an aggregate function",
+                ident
+            );
+        }
```

Comment:
TIOLI these could be slightly differented by using `ensure!(sql_func.over.is_none(), "...", ident)`

Reply:
Call me a Go programmer, but the `ensure!` macro does too much control flow for my taste.

Reply:
you do you.